tests/test_start_date.py

```diff
@@ -0,0 +1,155 @@
+"""
+Test that the tap respects the start date
+"""
+from singer.utils import strptime_to_utc
+from tap_tester import connections, menagerie, runner
+from base import MambuBaseTest
+
+class StartDateTest(MambuBaseTest):
+    """
+    Test that the tap respects the start date
+    """
+
+    first_sync_start_date = None
+    second_sync_start_date = None
+    first_sync_records = None
+    second_sync_records = None
+
+    @staticmethod
+    def name():
+        return "tap_tester_mambu_start_date_test"
+
+    def setUp(self):
+        self.first_sync_start_date = self.get_properties()['start_date']
+
+        self.second_sync_start_date = self.get_properties(
+            original_properties=False
+        )['start_date']
+
+    def untestable_streams(self):
+        return set([
+            "communications", # Need to set up Twilio or email server to send stuff
+            "installments",
+        ])
+
+    def get_replication_key_values(self, stream, records):
+        all_records = []
+        for record in records:
+            # Build the primary key for this record, maintaining the same order for the fields
+            record_rep_key = [record[field]
+                              for field in sorted(self.expected_replication_keys()[stream])]
+            # Cast to a tuple to make it hashable
+            all_records.append(record_rep_key[0])
+        return all_records
+
+    def do_verify_replication_key_values(self, stream_name, records, start_date):
+        rep_values = self.get_replication_key_values(stream_name, records)
+
+        for value in rep_values:
+            self.assertGreaterEqual(
+                strptime_to_utc(value),
+                strptime_to_utc(start_date)
+            )
+
+    def verify_replication_key_values(self, stream_name):
+        self.do_verify_replication_key_values(
+            stream_name,
+            self.first_sync_records,
+            self.first_sync_start_date
+        )
+        self.do_verify_replication_key_values(
+            stream_name,
+            self.second_sync_records,
+            self.second_sync_start_date
+        )
```

Comment:
I don't think the assertion methods add value to the tests. It is not clear to me wheat we are asserting by the method signature. I think the comments below are clear but I don't think this is a pattern we wan to use in tap-tester.